app/notifications/validators.py

```diff
@@ -42,7 +42,7 @@ def check_service_over_api_rate_limit(
     Raises:
         RateLimitError: If the service has exceeded its API rate limit.
     """
-    if current_app.config['API_RATE_LIMIT_ENABLED'] and current_app.config['REDIS_ENABLED']:
```

Comment:
I believe `API_RATE_LIMIT_ENABLED` needs to stay. It's used in the lower envs (dev & staging) to enable rate limiting, which is not imposed in perf & prod.